deploy/deployment-manager/deploy.sh

```diff
@@ -115,7 +116,7 @@ fi
 # Apply the deployment manager templates
 run_command "gcloud deployment-manager deployments create --automatic-rollback-on-error ${DEPLOYMENT_NAME} --project ${PROJECT_NAME} \
     --template compute_engine.py \
-    --properties elasticAgentVersion:${STACK_VERSION},fleetUrl:${FLEET_URL},enrollmentToken:${ENROLLMENT_TOKEN},allowSSH:${ALLOW_SSH},zone:${ZONE},elasticArtifactServer:${ELASTIC_ARTIFACT_SERVER},scope:${SCOPE},parentId:${PARENT_ID}"
+    --properties elasticAgentVersion:${STACK_VERSION},fleetUrl:${FLEET_URL},enrollmentToken:${ENROLLMENT_TOKEN},allowSSH:${ALLOW_SSH},zone:${ZONE},elasticArtifactServer:${ELASTIC_ARTIFACT_SERVER},scope:${SCOPE},parentId:${PARENT_ID},serviceAccountName:${SERVICE_ACCOUNT_NAME}"
 
```

Comment:
Will the `SERVICE_ACCOUNT_NAME` param be propagated from the UI and if not we'll create one?
In case the user provides a wrong SA name, the deployment will not succeed and an error will be printed to the user.
Are we ok with it? do we want to add validation that the SA exists and if not to create one?

Reply:
I'm also unsure about the case where a service account is not provided at all. In that case, it will default to serviceAccountName:false. Is this case considered?

Reply:
> Will the `SERVICE_ACCOUNT_NAME` param be propagated from the UI and if not we'll create one? 

no, `SERVICE_ACCOUNT_NAME` is not user-facing. 

the addition of `SERVICE_ACCOUNT_NAME` to the `deploy.sh` script is done purely for testing purposes of deploying an agent with an existing service account. it simulates the new agentless flow, where a user will deploy a service account and we'd use its key in the integration credentials input. 

to test the deployment of a service account is working properly, i've used the existing agent deployment script to make it so that it can operate either by deploying the full resources: VM + SA, or just a VM, while using an existing SA, which is currently passed only during tests. 

> In case the user provides a wrong SA name, the deployment will not succeed and an error will be printed to the user. Are we ok with it? do we want to add validation that the SA exists and if not to create one?

as mentioned, `SERVICE_ACCOUNT_NAME` is only used in tests, so we don't want to deploy a new service account if `SERVICE_ACCOUNT_NAME` doesn't exists. if for some reason `SERVICE_ACCOUNT_NAME` is provided but doesn't exists, failing the deployment would be the right outcome.